domain/src/test/java/org/oppia/domain/repository/SkillRepositoryTest.kt

```diff
@@ -0,0 +1,362 @@
+      assertThat(conceptCardView.skillId).isEqualTo(FRACTIONS_SKILL_ID_2)
+      assertThat(conceptCardView.skillThumbnail.thumbnailGraphic).isEqualTo(LessonThumbnailGraphic.MIXED_NUMBERS_AND_IMPROPER_FRACTIONS)
+    }
+
+  @Test
+  @ExperimentalCoroutinesApi
+  fun tesSkillSummary_storeAllSkillSummariesToProtoDatabase_getRatiosSkillID0_correctSkillSummaryData() =
+    runBlockingTest(coroutineContext) {
+      skillRepository.storeAllSkillSummariesToProtoDatabase()
+      advanceUntilIdle()
+
+      val skillSummaryDataProvider = skillRepository.getSkillSummaryDataProvider(RATIOS_SKILL_ID_0)
+      dataProviders.convertToLiveData(skillSummaryDataProvider)
+        .observeForever(mockConceptCardViewObserver)
+      advanceUntilIdle()
+
+      verifyGetConceptCardViewSucceeded()
+      val conceptCardView = conceptCardViewResultCaptor.value.getOrThrow()
+      assertThat(conceptCardView.skillId).isEqualTo(RATIOS_SKILL_ID_0)
+      assertThat(conceptCardView.skillThumbnail.thumbnailGraphic).isEqualTo(LessonThumbnailGraphic.DERIVE_A_RATIO)
+    }
+
+  @Test
+  @ExperimentalCoroutinesApi
+  fun tesSkillSummary_addSkillSummaryList_getSkillSummary0_correctSkillSummaryData() =
+    runBlockingTest(coroutineContext) {
+      skillRepository.addSkillSummaryList(createSkillSummaryList())
+      advanceUntilIdle()
+
+      val skillSummaryDataProvider = skillRepository.getSkillSummaryDataProvider(TEST_SKILL_ID_0)
+      dataProviders.convertToLiveData(skillSummaryDataProvider)
+        .observeForever(mockConceptCardViewObserver)
+      advanceUntilIdle()
+
+      verifyGetConceptCardViewSucceeded()
+      val conceptCardView = conceptCardViewResultCaptor.value.getOrThrow()
+      assertThat(conceptCardView.skillId).isEqualTo(TEST_SKILL_ID_0)
+      assertThat(conceptCardView.description).isEqualTo(DUMMY_DESCRIPTION_TEXT)
+      assertThat(conceptCardView.thumbnailUrl).isEqualTo(DUMMY_THUMBNAIL_URL)
+      assertThat(conceptCardView.skillThumbnail.thumbnailGraphic).isEqualTo(LessonThumbnailGraphic.ADDING_AND_SUBTRACTING_FRACTIONS)
+    }
+
+  @Test
+  @ExperimentalCoroutinesApi
+  fun tesSkillSummary_storeAllSkillSummariesToProtoDatabase_getInvalidSkillID_noSkillSummaryDataFound() =
+    runBlockingTest(coroutineContext) {
+      skillRepository.addSkillSummaryList(createSkillSummaryList())
+      advanceUntilIdle()
+
+      val skillSummaryDataProvider = skillRepository.getSkillSummaryDataProvider(INVALID_SKILL_ID)
+      dataProviders.convertToLiveData(skillSummaryDataProvider)
+        .observeForever(mockConceptCardViewObserver)
+      advanceUntilIdle()
+
+      verifyGetConceptCardViewSucceeded()
+      val conceptCardView = conceptCardViewResultCaptor.value.getOrThrow()
+      assertThat(conceptCardView).isEqualTo(ConceptCardView.getDefaultInstance())
+    }
+
+  @Test
+  @ExperimentalCoroutinesApi
+  fun tesSkillSummary_addSkillSummaryList_getSkillSummary1_correctSkillSummaryData() =
```

Comment:
Here & elsewhere: the test names might be a bit clearer if you write it like:

*testAction_withCondition_andCondition_hasOutcome*

E.g.:

``testGetSkillSummary_forExistingSummary_summaryWithDummyData_returnsCorrectSummary``

Or similar. Ditto elsewhere.